tests/testthat/test-contribution_matrix.R

```diff
@@ -374,3 +374,350 @@ test_that("CreateContributionMatrix() produces a matrix of the correct format wh
+                                                  basic_or_all_parameters = "basic",
+                                                  study_or_comparison_level = "comparison",
+                                                  absolute_or_percentage = "absolute",
+                                                  weight_or_contribution = "weight")
+  
+  expected_V <- matrix(c(2^2/50 + 2^2/51, 0, 0,
+                         0, 3^2/60 + 3^2/61, 3^2/60,
+                         0, 3^2/60, 3^2/62 + 3^2/60),
+                       byrow = TRUE, nrow = 3)
+  expected_lambda_tau <- matrix(c(0.9^2, 0, 0,
+                                  0, 0.9^2, 0.9^2/2,
+                                  0, 0.9^2/2, 0.9^2),
+                                byrow = TRUE, nrow = 3)
+  expected_V_star <- rbind(cbind(expected_V, matrix(rep(0, times = 9), nrow = 3)),
+                           cbind(matrix(rep(0, times = 9), nrow = 3), expected_lambda_tau))
+  
+  expected_X <- matrix(c(1, 0, 1-1.25,
+                         1, 0, 1.5-1.25,
+                         0, 1, 1.5-1.25),
+                       byrow = TRUE, nrow = 3)
+  expected_X_star <- rbind(cbind(diag(1, nrow = 3), matrix(rep(0, times = 9), nrow = 3)),
+                           cbind(diag(1, nrow = 3), -expected_X))
+  expected_XVX_matrix <- solve(t(expected_X_star)%*% solve(expected_V_star) %*% expected_X_star) %*% t(expected_X_star) %*% solve(expected_V_star)
+  expected_A_matrix <- expected_XVX_matrix[4:6, 1:3]
+  expected_contribution_matrix <- round(t(abs(expected_A_matrix)), digits = 2)
+  rownames(expected_contribution_matrix) <- rownames(contribution_matrix)
+  colnames(expected_contribution_matrix) <- colnames(contribution_matrix)
+  
+  expect_equal(contribution_matrix, expected_contribution_matrix)
+})
+
+
+
+test_that("CreateContributionMatrix() includes all parameters when 'basic_or_all_parameters' = 'all'", {
+  data <- read.csv("Cont_long_cont_cov_small.csv")
+  
+  treatment_ids <- list(Number = 1:3, Label = c("Hydrogen", "Neon", "Carbon"))
+  
+  contribution_matrix <- CreateContributionMatrix(data = data,
+                                                  treatment_ids = treatment_ids,
+                                                  outcome_type = "Continuous",                           
+                                                  outcome_measure = "MD",
+                                                  effects_type = "fixed",
+                                                  cov_parameters = "shared",
+                                                  basic_or_all_parameters = "all",
+                                                  study_or_comparison_level = "comparison",
+                                                  absolute_or_percentage = "absolute",
+                                                  weight_or_contribution = "weight")
+  
+  basic_contribution_matrix <- CreateContributionMatrix(data = data,
```

Comment:
`basic_contribution_matrix` isn't used anywhere in the test - remove?

Reply:
Correct, I had wanted to use it but there was a reason why I couldn't so had to do it manually. I think it was because it was impossible to tell which values had been positive and which negative before taking the absolute values.